### PR TITLE
Free MAGMA queues after use

### PR DESCRIPTION
**Pull request by vishwakftw**
This PR is a minor change, just adds a simple `magma_queue_destroy` function to the implementation of `Gesv`.

Also, I have replaced calls for obtaining handles with those already written in ATen.
```
THCState_getCurrentSparseHandle(at::globalContext().getTHCState()) --> getCurrentCUDASparseHandle()
THCState_getCurrentBlasHandle(at::globalContext().getTHCState()) --> getCurrentCUDABlasHandle()
```

cc: @soumith @zou3519

### CODE REVIEW

File: aten/src/ATen/native/cuda/Gesv.cu [hunk 1]
```diff
@@ -53,12 +53,16 @@ static magma_queue_t createMagmaQueue(const Tensor& tensor) {
...
-      THCState_getCurrentBlasHandle(context.getTHCState()),
-      THCState_getCurrentSparseHandle(context.getTHCState()),
+      at::cuda::getCurrentCUDABlasHandle(),
+      at::cuda::getCurrentCUDASparseHandle(),
       &magma_queue);
   return magma_queue;
 }
 
+static void destroyMagmaQueue(magma_queue_t& existing_queue) {
+  magma_queue_destroy(existing_queue);
```
[the remark this answers is not in the document]
Comment (vishwakftw) [reply aten/src/ATen/native/cuda/Gesv.cu]: Synchronization is not specifically specified in the examples, so I believe this is safe.

**Comment by fmassa** [comment]
I think this looks reasonable to me, but can anyone else have a look at this? cc @soumith

File: aten/src/ATen/native/cuda/Gesv.cu (hunk 1 above)
[the remark this answers is not in the document]
Comment (soumith) [reply aten/src/ATen/native/cuda/Gesv.cu]: any destroy if it calls cudaFree will implicitly sync. I dont think sync and destroy is needed.

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
fixed some merge conflicts, will wait for CI to finish

**Pull request closed without merging**